dbt/adapters/athena/impl.py

```diff
@@ -46,6 +46,13 @@ def convert_number_type(cls, agate_table: agate.Table, col_idx: int) -> str:
     def convert_datetime_type(cls, agate_table: agate.Table, col_idx: int) -> str:
         return "timestamp"
 
+    def drop_relation(self, relation):
```

Comment:
I think this is a good idea and if it works, it is perfect !

A thing I would try is to keep the drop_relation in python but try to rename the macro athena__drop_relation and try calling adapter.drop_relation in SQL macros.

This way, contributors can use the expected adapter.drop_relation and our login enables to :
- Call the right macro
- **WITH** the config

Reply:
Not sure if I fully get it, the `AthenaAdapter.drop_relation` should take priority  to everything right? but if so, how config is passed? e.g. drop_relation receive only relation, nothing more, so I believe that when the macro is called config is null, and again we have some issues in case of iceberg (again makeing clear that for iceberg we want to use simple drop statements, for not iceberg tables we want do drop the table from the db and purge s3 path).
Said so, I like this solution, but if we want to be more future proof there are some edge cases where this won't work.

I recommend here to use something that is being solved in dbt-glue.

In dbt-glue there is an overwrite for `glue__drop_relation`, as you can see in [here](https://github.com/aws-samples/dbt-glue/blob/main/dbt/include/glue/macros/adapters.sql#L27-L39), there is no config passed. The magic is on the `{% set rel_type = adapter.get_table_type(relation)  %}` that make a glue api call to pick the table type.
Based on the relation type then we can deal with it downstream, in our case pretty easy:
* if iceberg -> simple drop
* if external(hive) -> glue table drop and s3 purge.

Reply:
@nicor88 Yes, I understand your concerns regarding Iceberg vs. other tables and conditionally purging S3 paths. I also agree that your suggestion calling an adapter method seems like a clean way to do it. However, how do I get the model configs for `table_type` and `incremental_strategy` in the adapter? I guess I could find out if a table is of Iceberg type or not via a remote call (though wouldn't like that approach), but `incremental_strategy` is a user config. Where do I get it from?

I must say that I am at a complete loss how the `config` variable in macros is made available and why it doesn't contain all model configs. In my test project, it doesn't even matter if the macro is named `drop_relation` or `athena__drop_relation`, the config values for `table_type` and `incremental_strategy` are never present in the macro. I don't get how this ever worked if the `config` is not populated correctly.

Sorry for the rant, but could this simply be a bug in `dbt-core` that sometimes `config` does not contain all user-specified model configs?

Reply:
Indeed you are right, the issue is in dbt-core, somehow the config is not available in that context, that's why we are discussing workarounds here. I'm also lost on how the config is passed, me and @Jrmyy break our head on the same issue some weeks ago.

For what I can tell you we can complete ingore the config, as for dbt-athena we care only if the relation is iceberg or not, because based on that we need to do different type of dropping.